[28.x] deprecate NetworkSettingsBase, DefaultNetworkSettings

akerouanton opened the pull request:
- Supersedes https://github.com/moby/moby/pull/45945

**- What I did**

**api/t/ctr: deprecate NetworkSettingsBase.Bridge**

This field provides little value as it's only set when the daemon is started with --bridge flag specified. There's no equivalent field in NetworkSettings. Deprecate it.

**api/t/ctr: deprecate NetworkSettingsBase**

Most fields in NetworkSettingsBase are deprecated, so deprecate the whole struct. The few fields which aren't deprecated will move to the NetworkSettings struct in v29.

Mark all deprecated fields as `omitempty`.

**api/t/ctr: deprecate DefaultNetworkSettings**

This struct is only used to report the networking state for the default bridge network when the container is connected to it.

It was deprecated in v1.09 (API v1.21) but has never been removed since then. Unfortunately, the deprecation warning was wrongly formatted.

Fix the deprecation warning.

**- Human readable description for the release notes**

```markdown changelog
- Go-SDK: Deprecate field `NetworkSettingsBase.Bridge`, struct `NetworkSettingsBase`, all the fields of `DefaultNetworkSettings`, and struct `DefaultNetworkSettings`.
```

akerouanton committed:
api/t/ctr: deprecate NetworkSettingsBase.Bridge

This field provides little value as it's only set when the daemon is
started with --bridge flag specified, and the inspected container is
connected to the default bridge network.

Unfortunately, there's no equivalent field in NetworkSettings.Networks.

Signed-off-by: Albin Kerouanton <albinker@gmail.com>
(cherry picked from commit 16dc39136cae7f04b04f8fc50ac312d78a4c79d5)
Signed-off-by: Albin Kerouanton <albinker@gmail.com>

akerouanton committed:
api/t/ctr: deprecate NetworkSettingsBase

Most fields in NetworkSettingsBase are deprecated, so deprecate the
whole struct. The few fields which aren't deprecated will move to the
NetworkSettings struct in v29.

Signed-off-by: Albin Kerouanton <albinker@gmail.com>
(cherry picked from commit 80bb864fd611bb1a065017738e1e86dbec2fa921)
Signed-off-by: Albin Kerouanton <albinker@gmail.com>

akerouanton committed:
api/t/ctr: deprecate DefaultNetworkSettings

This struct is only used to report the networking state for the default
bridge network when the container is connected to it.

It was deprecated in v1.09 (API v1.21), and scheduled for removal in
v1.11. Unfortunately, the deprecation warning was wrongly formatted in
the Go code. However, deprecation warnings are already present in
swagger.yaml, so don't touch it.

Signed-off-by: Albin Kerouanton <albinker@gmail.com>
(cherry picked from commit b7c597ec35a6470483628539838c232212751dc4)
Signed-off-by: Albin Kerouanton <albinker@gmail.com>

akerouanton committed:
testutil/daemon: remove unused FindContainerIP

Signed-off-by: Albin Kerouanton <albinker@gmail.com>
(cherry picked from commit d0de29351348a04f16505341593cd57f2640a018)
Signed-off-by: Albin Kerouanton <albinker@gmail.com>

akerouanton commented on docs/api/version-history.md:
```diff
@@ -20,6 +20,17 @@ keywords: "API, Docker, rcli, REST, documentation"
+  `NetworkSettings.MacAddress`.
+* `GET /containers/{id}/json` now omits the following `NetworkSettings` fields when
+  empty: `Bridge`, `HairpinMode`, `LinkLocalIPv6Address`, `LinkLocalIPv6PrefixLen`,
+  `SecondaryIPAddresses`, `SecondaryIPv6Addresses`.
```
Done.

thaJeztah commented on api/types/types_deprecated.go:
```diff
@@ -46,15 +46,14 @@ type NetworkSettings = container.NetworkSettings
 //
-// Deprecated: use [container.NetworkSettingsBase].
-type NetworkSettingsBase = container.NetworkSettingsBase
+// Deprecated: use [container.NetworkSettings].
```
Was this intentionally pointing to the other type? (not "base")?

akerouanton committed:
api/types: update deprecation notice on NetworkSettingsBase

container.NetworkSettingsBase is deprecated in v28.4, and scheduled for
removal in v29. However, it contains three fields that aren't deprecated
and that will be moved to NetworkSettings.

Update the deprecation notice on NetworkSettingsBase to advise users
to access NetworkSettingsBase's fields through NetworkSettings.

Signed-off-by: Albin Kerouanton <albinker@gmail.com>
